Feature/#6 footer공통컴포넌트작업

Stemplun opened the pull request:
### 이슈 번호

close #6 

### 작업 사항 요약

웹사이트 하단에 사용할 Footer 공통 컴포넌트를 생성하였습니다. 

### 작업 결과
![image](https://github.com/user-attachments/assets/8f99ac82-ad8c-418a-b938-cbe18c6bd3e0)
페이지에 Footer 컴포넌트를 적용을 하면 이미지와 같이 나옵니다. 

![image](https://github.com/user-attachments/assets/931d4ecd-dcbe-45ae-863d-78b7445cabbf)
우측에 보이는 이 이미지들은 따로 sns에 연결을 하지 않아도 될 것 같아 button이 아니라 Image컴포넌트를 사용하였습니다.

Stemplun committed:
Footer에 사용할 sns 이미지 다운

Stemplun committed:
Footer 컴포넌트 추가

y00n1verse commented on src/app/_component/footer.tsx:
```diff
@@ -0,0 +1,27 @@
+import Image from 'next/image';
+export default function Footer() {
```
🔨 impot문과  함수 컴포넌트 사이 한 줄 띄우면 구분이 잘 되어 더 깔끔할 것 같아요!

Stemplun replied:
바로 적용 하겠습니다!

y00n1verse commented on src/app/_component/footer.tsx:
```diff
@@ -0,0 +1,27 @@
+      <div className="flex justify-around gap-2">
+        <Image
+          src={'/image/envelope-square.svg'}
+          alt=""
```
🔨 페이스북, 인스타그램 등과 같이 각각 alt 속성을 작성하면 접근성에 더 좋을 것 같아요!

Stemplun replied:
바로 적용 하겠습니다!

y00n1verse commented on tsconfig.json:
```diff
@@ -1,7 +1,11 @@
   "compilerOptions": {
     "target": "ES2017",
-    "lib": ["dom", "dom.iterable", "esnext"],
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
```
❓ 오잉 같은 내용인데 커밋에 남네요! 줄바꿈 기준이 다른 걸까요?

Stemplun replied:
아마 prettier 설정 때문에 그렇게 된게 아닌가 싶습니다.

y00n1verse commented on src/app/_component/footer.tsx:
```diff
@@ -0,0 +1,27 @@
+          width={24}
+          height={24}
+        />
+        <Image
+          src={'/image/facebook-square.svg'}
+          alt=""
+          width={24}
+          height={24}
+        />
```
이미지와 속성 잘 지정해주셨네요! 👍

y00n1verse reviewed (approved):
굉장히 빠르게 만들어주셨네요! ㅎㅎ 

디자인 컴포넌트라 기능적인 부분에서는 문제가 없을 것 같아요. 제 사소한 코멘트만 참고해주시면 감사하겠습니다. 수고하셨습니다! 👍

Stemplun committed:
Footer PR 1차 수정

- import문과 아래 코드들이 붙어 있어 가독성이 떨어지는 점을 두 코드 사이에 공간을 띄어두는 것으로 가독성을 높였습니다.
- Image 컴포넌트의 alt속성에 해당 이미지의 설명을 추가하여 접근성을 높였습니다.

Hogn-hyeonbin reviewed (approved):
확인했습니다 되게 빠르게 작업해 주셨네용 수고하셨습니다!

swp91 reviewed (approved):
하림님이 작업 스타트를 끊어주셨군요 감사합니다 고생하셨습니다!!👍👍
지윤님 리뷰 너무 꼼꼼하시구 현빈님도 멘트 감사합니다
역시 한명한명이 정예인 우리팀 너무 좋아요

Stemplun committed:
반응형 추가

- 375px(모바일)버전에서는 구조가 변경이 되어 하나의 코드로는 하기 힘들다 판단했습니다.
- 그래서 useState와 useEffect를 활용하여 현재 윈도우의 width를 측정하여 640px을 기준으로 각각 다른 코드를 작성하였습니다.
